### DISCUSSION TITLE
Intermittent client freeze

### DISCUSSION THREAD

**Discussion by decenttp** [ltsp/ltsp#560] [Q&A] [answered]
Hi,

I have around 10 clients (All clients have over 4 GB RAM) connected to the LTSP server ( Intel(R) Xeon(R) CPU E3-1225 v5 @ 3.30GHz with 16 gigs of ram) but for some reason they freeze or hang up intermittently. How can i find the reason why it's happening and resolve this. I tried adding an additional gigabit network card and created a bond to improve network performance but that did not stop client freeze.

Thanks

**Comment by alkisg** [comment]
Network speed shouldn't matter wrt freezing.
Two good suspects are "graphics card driver issues" or "out of memory conditions".
You can check the first one with any live CD or local linux distribution; try to use the exact same kernel though as these vary per kernel/xorg version.
To check if it's the second one, if they have some local disk, create a swap partition there, or if they don't, try to create network swap: https://github.com/ltsp/ltsp/issues/7#issuecomment-915033243

**Reply by decenttp** [reply]
I started off with creating a network swap option following your advice from [#7 (comment)](https://github.com/ltsp/ltsp/issues/7#issuecomment-915033243) . I am not sure however it's taking effect. Do i need to do something else too?

I did run  :

ltsp initrd 

I had also added following parameters in ltsp.conf : 
IMAGE_TO_RAM=0 // I intended to test with this option disabled
DEFAULT_IMAGE="Zorin-Custom" 

It didn't work however as during client boot the default image is still different.

Thanks

**Comment by decenttp** [comment]
Thanks for your response. I have tested with 3 different boot image options. One the default Ubuntu mate option that comes from the server image (Ubuntu 20.04) and two others, 1 using Q4OS and 1 using Zorin Lite to ensure low resource requirements. I would test for both the potential issues you have mentioned along with the solutions and report back.

Thanks

**Comment by alkisg** [comment]
To see if you activated swap, run `free`.
DEFAULT_IMAGE goes in ltsp.ipxe so it takes effect after `ltsp ipxe`.
IMAGE_TO_RAM=1 needs a lot of RAM, it's better not to use it.

**Reply by decenttp** [reply]
> To see if you activated swap, run `free`. DEFAULT_IMAGE goes in ltsp.ipxe so it takes effect after `ltsp ipxe`. IMAGE_TO_RAM=1 needs a lot of RAM, it's better not to use it.

I can see swap on running free but I meant checking on the network swap if it was any different. I am using IMAGE_TO_RAM=0 to disable usage of RAM. Is 0 the default for IMAGE_TO_RAM?

Clients continue to hang still frequently.

**Comment by alkisg** [comment]
`cat /proc/swaps` will tell you where your swap is coming from.
IMAGE_TO_RAM is 0 by default, yes.

**Reply by decenttp** [reply]
cat /proc/swaps on the ltsp server shows that currently only /swapfile location is being used. Please see attached. 

![image](https://user-images.githubusercontent.com/88873876/136511758-a7b04d64-d54b-43b0-b8c7-08766d6cc365.png)

It does not seem like network swap is working for the LTSP clients. I have also attached my current ltsp.conf and local.conf

[](url)
[ltsp.zip](https://github.com/ltsp/ltsp/files/7309172/ltsp.zip)
[local.zip](https://github.com/ltsp/ltsp/files/7309173/local.zip)

**Comment by alkisg** [comment]
Run `cat /proc/swaps` on the ltsp client, not on the ltsp server.

**Reply by decenttp** [reply]
My bad. I ran it on clients now and does not seem like it is working: 

user1@ltsp220:~/Desktop$ cat /proc/swaps
Filename				Type		Size	Used	Priority

**Comment by alkisg** [comment, marked as answer]
Yeah this means network swap wasn't configured successfully.
If you want to come to IRC we could review this live: https://ltsp.org/advanced/chat-room/

You could also start your tests with the other ideas, "local swap" or "live cd to test the graphics drivers".

**Reply by decenttp** [reply]
Joining on IRC

**Reply by decenttp** [reply]
Local swap resolved the issue. I had implemented local swap as per your recommendation on all of the clients and that has resolved the issue.

**Comment by alkisg** [comment]
Great! Let's hope the new [systemd-oomd](https://man.archlinux.org/man/systemd-oomd.service.8.en) will make "out of RAM handling" better for Linux users, so that e.g. a message appears, a memory hungry app like the browser is closed, and the rest of the programs continue executing without issues.
